dbms/src/Storages/Page/V3/PageStorageImpl.cpp

```diff
@@ -168,7 +176,12 @@ PageMap PageStorageImpl::readImpl(NamespaceId ns_id, const PageIds & page_ids, c
     else
     {
         auto [page_entries, page_ids_not_found] = page_directory->getOrNull(page_id_v3s, snapshot);
-        auto page_map = blob_store.read(page_entries, read_limiter);
+        PageMap page_map = {};
+        if (!page_entries.empty())
+        {
+            page_map = blob_store.read(page_entries, read_limiter);
+        }
```

Comment:
Why do we need to add these checks?

Reply:
1. `page_entries` may empty
2. If `page_entries` is empty no need read from `BlobStore`
3. also If we put an empty `writebatch` call read(field) from `BlobStore`, it will get an exception. So, to unify this behavior, in mix mode, we don't let the `BlobStore` read empty entries.

Reply:
It is strange that the caller takes care of whether the input param is empty or not, normally we do that check inside the function. What about put those `if (xx.empty()) { xxx }` inside `BlobStore::read(...)`?

Reply:
`BlobStore` handles read(wb size) is perfect but a little differently.
- If the caller used to read(fields) with size 0, it will throw an exception to avoid read null from field offset.
- other read methods, will return an empty map for that.

IMO, The problem is not "the caller takes care of whether the input param is empty or not", But "the caller knows it may be a null arg input, don't waste time to call for that"? 

So I think it better to add a check to avoid empty args:
- avoid read(fields) and get an exception.
- I know it may be empty, so check if it is empty or not, If empty, no need to read.

is that make sense to you?